### DISCUSSION TITLE
ROS + VSCode 编程环境快速搭建 | Longlin's Blog

### DISCUSSION THREAD

**Discussion by giscus** [longlin10086/blog#7] [Blog comments]
# ROS + VSCode 编程环境快速搭建 | Longlin's Blog

在自己的电脑上开发数图实验

https://blog.longlin.tech/posts/digital-image/

**Comment by MaxwellJay256** [comment]
如果你在输入 `catkin build` 后系统提示

```bash
catkin：未找到命令
```

这是因为 `catkin build` 不是默认安装的工具。要使用 `catkin build` 命令，需要单独安装 `catkin-tools` 包。

```bash
sudo apt-get install python3-catkin-tools
```

`catkin build` 与 `catkin_make` 是有一些区别的。
1. `catkin_make` 是一个简化的命令行工具，主要围绕 CMake 和 make 命令进行操作。它将整个工作空间视为单个 CMake 项目，这使增量构建更快，但如果需要重新生成构建文件，构建速度会变慢。
2. `catkin build` 类似于 `catkin_make_isolated`, 它独立地在工作空间的源空间中构建每个包，以防止构建时的串扰。

**Comment by kowyo** [comment]
可以试试 vscode-rog 这个插件，可以自动创建 `.vscode/c_cpp_properties.json` 文件
